cpp/dolfin/io/XDMFFile.cpp

```diff
@@ -1391,39 +1392,72 @@ mesh::Mesh XDMFFile::read_mesh(MPI_Comm comm,
   assert(gdims.size() == 2);
   assert(gdims[1] == gdim);
 
-  // Geometry
-  const auto geometry_data = xdmf_read::get_dataset<double>(
-      _mpi_comm.comm(), geometry_data_node, parent_path);
-  const std::size_t num_local_points = geometry_data.size() / gdim;
-
-  Eigen::Map<const EigenRowArrayXXd> points(geometry_data.data(),
-                                            num_local_points, gdim);
   // Get topology dataset node
   pugi::xml_node topology_data_node = topology_node.child("DataItem");
   assert(topology_data_node);
-
-  // Topology
   const std::vector<std::int64_t> tdims
       = xdmf_utils::get_dataset_shape(topology_data_node);
-  const auto topology_data = xdmf_read::get_dataset<std::int64_t>(
-      _mpi_comm.comm(), topology_data_node, parent_path);
   const std::size_t npoint_per_cell = tdims[1];
-  const std::size_t num_local_cells = topology_data.size() / npoint_per_cell;
-  Eigen::Map<const EigenRowArrayXXi64> cells(topology_data.data(),
-                                             num_local_cells, npoint_per_cell);
 
-  // Set cell global indices by adding offset
-  const std::int64_t cell_index_offset
-      = MPI::global_offset(_mpi_comm.comm(), num_local_cells, true);
-  std::vector<std::int64_t> global_cell_indices(num_local_cells);
-  std::iota(global_cell_indices.begin(), global_cell_indices.end(),
-            cell_index_offset);
+  if (MPI_COMM_NULL != comm)
```

Comment:
Let's just check that `comm != MPI_COMM_NULL` and throw an error. It is up to the caller to supply a valid comm, so this should just be a check.

Reply:
I will add the comment:

// If this rank isn't in the (new) communicator, it will be MPI_COMM_NULL.